### PR TITLE
feat: Added badge endpoint for Jenkins jobs

### PR DESCRIPTION
**Pull request by deepak-bhardwaj-ps**
Added a new badge endpoint for Jenkins jobs.

### CODE REVIEW

**Comment by deepak-bhardwaj-ps** [comment]
@amio please can you look into this PR

**Comment by amio** [comment]
Sorry for missing this @deepak-bhardwaj-ps

File: api/jenkins.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  return hours < 2 ? "green" : hours < 6 ? "orange" : "red";
+};
+
+async function jenkinsLastBuild({ hostname, job }: PathArgs) {
+  const endpoint = `https://${hostname}/${job}/lastBuild/api/json?tree=result,timestamp,estimatedDuration`;
+  return await got(endpoint).json<any>();
+}
+
+async function jenkinsBuilds({ hostname, job }: PathArgs) {
+  const endpoint = `https://${hostname}/${job}/api/json?tree=builds[number,status,timestamp,id,result]`;
```
Comment (amio) [inline api/jenkins.ts:57]: Do we need to handle paging here? Would it return a quite large json if there's a lot builds?

File: api/jenkins.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      h: () => "h",
+      m: () => "m",
+      s: () => "s",
+      ms: () => "ms",
+    },
+  },
+});
+
+const statusToColor = (status: string) => {
+  return status.toUpperCase() == "SUCCESS" ? "green" : "red";
```
Comment (amio) [inline api/jenkins.ts:41]:
```suggestion
  return status.toUpperCase() === "SUCCESS" ? "green" : "red";
```

File: api/jenkins.ts (hunk 1 above)
Comment (deepak-bhardwaj-ps) [reply api/jenkins.ts:57]: We can; however, the default behaviour of API restrict the response to 100 entries.

**Comment by amio** [comment]
Nice work, thanks! @deepak-bhardwaj-ps

**Pull request merged**